### PR TITLE
feat: update the Dockerfiles to `v2.2`

### PR DESCRIPTION
**Pull request by ottobolyos**

### CODE REVIEW

**Review by wsobel — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,6 @@
 # MTConnect Public C++ Agent Docker image build instructions
 
+# TODO: Do we need notes like this to be in `Dockerfile`? I think the docs might be a better place for this. IMHO most notes in this file should be moved to the docs.
```
Comment (ottobolyos) [inline Dockerfile:3]: I wanted to remove these `TODO` comments before merging PR, but after discussing them further.

@wsobel, what do you think about moving these `notes` to the docs? :thinking:

File: Dockerfile [hunk 2]
```diff
@@ -27,90 +29,127 @@
...
 # ---------------------------------------------------------------------
 # build
 # ---------------------------------------------------------------------
 
 FROM os AS build
 
+# limit cpus so don't run out of memory on local machine
+# symptom: get error - "c++: fatal error: Killed signal terminated program cc1plus"
+# can turn off if building in cloud
+ARG CONAN_CPU_COUNT=10
```
Comment (ottobolyos) [inline Dockerfile:43]: We definitely need to change this to `2`. (I should have mark the PR as a draft :man_facepalming:.)

File: Dockerfile [hunk 3]
```diff
@@ -27,90 +29,127 @@
...
+  && conan create . \
+    --build=missing \
+    -c "tools.build:jobs=$CONAN_CPU_COUNT" \
+    -c tools.build:skip_test=True \
+    -o agent_prefix=mtc \
+    -o cpack=True \
+    -o "with_ruby=$WITH_RUBY" \
+    -o zip_destination=/root/agent \
+    -pr "$CONAN_PROFILE" \
+    -tf ''
```
Comment (ottobolyos) [inline Dockerfile:90]: I presume you have verified this is what you want to do in `Dockerfile` to build the agent.

File: Dockerfile [hunk 4]
```diff
@@ -27,90 +29,127 @@
...
 
 # ---------------------------------------------------------------------
 # release
 # ---------------------------------------------------------------------
 
 FROM os AS release
 
-LABEL author="mtconnect" description="Docker image for the latest Production MTConnect C++ Agent"
+# TODO: How about shortening the description to MTConnect Agent or at least MTConnect C++ Agent?
```
Comment (ottobolyos) [inline Dockerfile:98]: What do you think about shortening the description?

File: Dockerfile [hunk 5]
```diff
@@ -27,90 +29,127 @@
...
+LABEL author='mtconnect' description='Docker image for the latest Production MTConnect C++ Agent'
+
+ARG BIN_DIR='/usr/bin'
+# FIXME: `/etc` is not the best place for the app data.
+ARG MTCONNECT_DIR='/etc/mtconnect'
```
Comment (ottobolyos) [inline Dockerfile:103]: This might require some further discussion.

**Review by ottobolyos — COMMENTED** [review]
I suggest to revert the merge commit (if possible) and fix the issues. I have forgot to mark the PR as a draft and expected some further discution with @wsobel, as well as some more testing.

**Comment by wsobel** [comment]
It’s fine to merge since we are not tagging as a release yet. We can work on it in a branch.

**Comment by wsobel** [comment]
Can we work off a branch in the repo instead of your fork?

**Comment by ottobolyos** [comment]
> It’s fine to merge since we are not tagging as a release yet. We can work on it in a branch.
>
> Can we work off a branch in the repo instead of your fork?

Oh, okay. I that is the case, then yes, we can do it anywhere and anyhow you want. Sorry for the noise. :wink:

I thought you try to works the PRs in the fork branches until they are done and reviewed, and then merge them into the main branch (a different workflow).

**Comment by ottobolyos** [comment]
And could we create a draft PR from `update_docker_file_for_2_2` please? Unless it’s a problem for you, I prefer reviewing the changes in a PR while we work on it. Thanks! :pray:

**Comment by wsobel** [comment]
I created a branch: [update_docker_file_for_2_2](https://github.com/mtconnect/cppagent/tree/update_docker_file_for_2_2) to begin work on Docker-related issues. We are working on the repo branches until we are complete and then merge them in. Just makes it a bit easier for me and others to collaborate. 

The merge was a bit pre-mature, but I wanted to get it into another branch and didn't see an easy way to achieve this.

**Comment by wsobel** [comment]
Done: https://github.com/mtconnect/cppagent/pull/306

**Pull request merged**